math/wxMaxima/Portfile

```diff
@@ -1,112 +1,84 @@
-                    Provides the development version, which is typically updated every day.
-
-    # git describe --tags : Version-16.12.x-1834-gb32fed0
-    github.setup    andrejv wxmaxima b32fed0d535c05d589d1f573d2c0f4f839180d18
-    checksums       rmd160  86fcf85ddae42e84884a708d55acff393152ed23 \
-                    sha256  e34256294c51a37d80947b92d3cd979456bf0c8ae53a21ce6cbab63d99143bf2 \
-                    size    12739940
-    version         18.02-dev-20180616
-    revision        1
-
-    patchfiles      patch-devel-src_Dirstructure.cpp.diff
-    post-patch {
-        reinplace -W ${worksrcpath} "s|@@PREFIX@@|${prefix}|g" \
-                                    src/Dirstructure.cpp
-    }
-
-    livecheck.type  none
-}
+long_description    Maxima is a Computer Algebra System (CAS) and wxMaxima is a work book style \
+                    graphical front end for it based on wxWidgets
+                    
 
-if {${subport} eq ${name}} {
-    conflicts       wxMaxima-devel
-    long_description ${description}: \
-                    Provides the release version, which is typically updated every four months.
-
-    # get the source tarball from github.
-    github.setup    andrejv wxmaxima 18.02.0 Version-
-    checksums       rmd160  d1017d56bfd76e9c51dcc3230be89ebca1724560 \
-                    sha256  77f0414be54d8ea15acedfc3cedf72a57cc2b3ba5d4835a53f3009e050c05483 \
-                    size    10131293
-    revision        1
-
-    patchfiles      patch-src_Dirstructure.cpp.diff
-    post-patch {
-        reinplace -W ${worksrcpath} "s|@@PREFIX@@|${prefix}|g" \
-                                    src/Dirstructure.cpp
-    }
-    
-    livecheck.regex Version-(\[a-z0-9.\]+)${extract.suffix}
-}
 
-universal_variant   no
+checksums           rmd160  0c10602f6f1d777ff16fe85500f57222eb07fc27 \
+                    sha256  055ff34eb98539716a18f7cc11861b5a790fde46aaafabb68f1e690cb649d9df \
+                    size    15314478
 
-variant wx32 description {Use wxWidgets 3.2 - experimental!} {}
+livecheck.regex     Version-(\[a-z0-9.\]+)${extract.suffix}
 
-if {[variant_exists wx32] && [variant_isset wx32]} {
-    # we want to use the wxWidgets-3.2
-    wxWidgets.use   wxWidgets-3.2
-} else {
-    wxWidgets.use   wxWidgets-3.0
-}
+universal_variant   no
+
+# Note: as of Match 2020 (1b00466)the build with wxWidgets-3.2 has serious issues with scrolling
+wxWidgets.use       wxWidgets-3.0
 
-depends_lib         port:${wxWidgets.port}
+depends_lib-append  port:${wxWidgets.port} \
+                    lib:${prefix}/lib/libomp/libomp:libomp
 
-depends_build       path:bin/cmake:cmake
+# Note: the advantage of path:bin/cmake:cmake over port:cmake is that it also allows e.g. cmake-devel
+# Note: the cmake port group seems to have issues with application packages
+depends_build-append path:bin/cmake:cmake
 
-depends_run         bin:maxima:maxima
+depends_run-append  port:gnuplot \
+                    port:maxima
```

Comment:
Can maxima-devel also be allowed? If so, you need to change it to a bin type dependency like for the cmake dependency.

Reply:
I think it is more likely that developers would have a maxima in home - maxima is mostly lisp and interpreted or built on demand. wxMaxima allows to choose a different maxima and maxima allows to choose a different lisp library folder. Developers know how to handle that. For the non-expert user I want to to have as few options as possible. For libomp and cmake  it makes sense because some people might need devel for something unrelated, but maxima and wxMaxima are more of a unit and it is not likely that some other packages requires maxima-devel as dependency.

Reply:
No port should require a dependency on a `-devel` port. Since one cannot have both installed, it would be very inconvenient for someone wanting to test `maxima-devel`. By default, it will install `maxima` for any user. If `maxima-devel` can work, it is preferred to allow it with a bin (in this case) dependency. Also it was the previous setting and I don't think we should change that.

Reply:
Sorry, I am stubborn about this. My point is that:
- Maxima changes more frequently than wxMaxima does and saying "can work with -devel" means that I should actually test this. We are not talking about an arbitrary minimally used library but a very complex command line tool and its GUI front end. I won't state that anything works here without testing it.
- I don't see any purpose in it. If Maxima-devel would be GIT master (which it is not), this would be a different thing. Developers would have an easy way to ask users to try latest master to test if something is fixed. But I don't see any value in supporting a four month old arbitrary development snapshot.
- Developers generally need very specific versions and know how to install these.
- Features which have no use do more harm than good because they mislead

In summary it would be extra testing effort to put it in without giving any advantage I see. If there would be a "Maxima-git-master" I would support it (also making clear that there is obviously no guarantee that this works) because this would have a value.

> No port should require a dependency on a -devel port

Then I am inclined to remove the support for openmp-devel. What purpose does it serve to support this? Only developers need to test such things and they should know how to put in an arbitrary openmp as openmp plain. At least this is what I did when testing this and it was easy enough to setup a local MacPorts patch repo. The likelihood that an arbitrary development snapshot of openmp does serve anybody any purpose is really pretty slim. If someone has a good reason to test that specific version (s)he has 99.9% likely good reason to test a bunch of other versions as well and anyway needs to setup a local port repo. And the remaining 0.1% of an already small number of people rounded to nearest integer is very likely 0.

Reply:
P.S.: to give a concrete example let's look at openmp-devel and my wxMaxima port. The pure existence of this did lead to:

- I spent ~45 minutes trying if it would fix my issues, which it didn't (the issue was that build options where missing)
- I forgot to remove the -devel, so that you had to comment on this in the review and I had to fix it
- If your review wouldn't have been so careful this might have slipped, so that wxMaxime would have dependend on openmp-devel which might have lead to all sorts of trouble with people using openmp for other stuff (which is not likely since openmp and Maxima are mostly used by scientist doing computations)
- It is most likely that everybody else having issues with openmp on OSX has the same experience since openmp is very special (it requires support in the compiler which Xcode by default does not offer) and the -devel variant doesn't help with this.

So the pure existence of the -devel variant did lead to quite a bit of trouble and it is purely due to your good review that it did not lead to much more trouble. That's the reason why I am not a big fan of it.

Reply:
Since you are becoming maintainer (thank you!) it is your prerogative.